MapboxMobileEvents/MMEEvent.m

```diff
@@ -3,305 +3,336 @@
 #import "MMEConstants.h"
 #import "MMECommonEventData.h"
 #import "MMEReachability.h"
+#import "MMEEventsManager.h"
 
 #if TARGET_OS_IOS || TARGET_OS_TVOS
-#import <UIKit/UIKit.h>
+#import "UIKit+MMEMobileEvents.h"
 #endif
 
+@interface MMEEvent ()
+@property(nonatomic,retain) MMEDate *dateStorage;
+@property(nonatomic,retain) NSDictionary *attributesStorage;
+
+@end
+
+#pragma mark -
+
 @implementation MMEEvent
 
-+ (instancetype)eventWithDate:(MMEDate *)eventDate name:(NSString *)eventName attributes:(NSDictionary *)attributes {
-    return [MMEEvent.alloc initWithDate:eventDate name:eventName attributes:attributes];
++ (NSDictionary *)nilAttributes {
+    static NSDictionary *nilAttributes = nil;
+    if (!nilAttributes) {
+        nilAttributes = @{};
+    }
+    return nilAttributes;
 }
 
-+ (instancetype)eventWithName:(NSString *)eventName attributes:(NSDictionary *)attributes {
-    return [MMEEvent eventWithDate:MMEDate.date name:eventName attributes:attributes];
+#pragma mark - Generic Events
+
++ (instancetype)eventWithAttributes:(NSDictionary *)attributes {
+    NSError *eventError = nil;
+    MMEEvent *newEvent = [MMEEvent eventWithAttributes:attributes error:&eventError];
+    if (eventError != nil) {
+        [MMEEventsManager.sharedManager reportError:eventError];
```

Comment:
Ok, the `sharedManager` issue is a problem, we rely on it for telemetry metrics collection and might be missing a lot of those due to how it's used in the maps SDK right now. The example in the Readme is wrong, the correct way to initialize the class (since moving over to relying on the singleton internally) is:

    [MMEEventsManager.sharedManager initializeWithAccessToken:userAgentBase:hostSDKVersion:]

In `MGLMapboxEvents.m:48` a new instance is created when the shared instance should be used.

Reply:
I've updated the readme and will make the change in the MapsSDK when I open the integration ticket.